### ISSUE TITLE
Proxy with authentcation is not accepted

### ISSUE THREAD

**Issue by OndraM** [mozilla/geckodriver#1872]
## System

* Version: 0.29.1
* Platform: arch linux 64
* Firefox: 87.0
* Selenium: [php-webdriver](https://github.com/php-webdriver/php-webdriver)

## Testcase

I initialize session from [php-webdirver](https://github.com/php-webdriver/php-webdriver), but I believe the issue does not depend on the language bindings used. So here they are just for reference and as an example.

```php
<?php

use Facebook\WebDriver\Remote\DesiredCapabilities;
use Facebook\WebDriver\Remote\RemoteWebDriver;
use Facebook\WebDriver\Remote\WebDriverBrowserType;
use Facebook\WebDriver\Remote\WebDriverCapabilityType;

require_once('vendor/autoload.php');

$capabilities = new DesiredCapabilities(
    [
        'browserName' => 'firefox',
        'proxy' => [
            'proxyType' => 'manual',
            'httpProxy' => 'user:pass@localhost:8080',
            'sslProxy' => 'user:pass@localhost:8080',
        ]
    ]
);
$driver = RemoteWebDriver::create('http://localhost:4444', $capabilities);
$driver->get('https://google.com/');
```

According to [W3C WebDriver specification](https://w3c.github.io/webdriver/#proxy), this form is a [proper way](https://w3c.github.io/webdriver/#dfn-host-and-optional-port) how to pass proxy credentials.

## Stacktrace

Output of `geckodriver --log debug` when script above is executed:

```
$ geckodriver --log debug
1618399457852   geckodriver     INFO    Listening on 127.0.0.1:4444
1618399458972   webdriver::server       DEBUG   -> POST /session {"capabilities":{"firstMatch":[{"browserName":"firefox","proxy":{"proxyType":"manual","httpProxy":"user:pass@localhost:8080","sslProxy":"user:pass@localhost:8080"}}]}}
1618399458972   webdriver::server       DEBUG   <- 400 Bad Request {"value":{"error":"invalid argument","message":"httpProxy is not of the form host[:port]: user:pass@localhost:8080","stacktrace":""}}
```

From the error ``httpProxy is not of the form host[:port]`` it seems geckodriver does not accept the W3C WebDriver form according to the specification.

In chromedriver this does work as expected.

**Comment by whimboo** [comment]
A proxy with authentication is not supported yet. The required work is tracked via https://bugzilla.mozilla.org/show_bug.cgi?id=1395886.

**Comment by 3052** [comment]
I need this, and its even worse because with Windows Firefox `https_proxy` is not recognized, so I have essentially no way to automate setting the proxy - which means I would have to manually set the proxy every time, or at a minimum the proxy credentials

**Comment by whimboo** [comment]
@3052 what do you mean with `https_proxy`? There is no such capability for WebDriver. As the example of the initial comment contains it is `sslProxy`.

Update: Lets actually follow-up on the bug given that you commented there as well.

**Comment by 3052** [comment]
> There is no such capability for WebDriver

yes I am well aware - but since `sslProxy` does not support authentication, I was looking for any solution, only to discover that Firefox Windows ALSO does not support the common environment variable `https_proxy`, so I have no way to automate setting proxy with authentication
